### ISSUE TITLE
Segmentation fault in docker container

### ISSUE THREAD

**Issue by meghma** [dotnet/SqlClient#1480]
### Describe the bug

I got the update 4.0.1 but I am still getting segmentation fault in docker container when starting my application.
This is my original ticket: https://github.com/dotnet/aspnetcore/issues/39253

If you are seeing an exception, include the full exceptions details (message and stack trace).

```
Exception message:
Starting application
Building host
2022-01-18 09:14:12.86315-05:00|INFO|Elliott.CollateralMgmt.Api.Service.ApiService.Start||Running host|
2022-01-18 09:14:12.94390-05:00|DEBUG|Microsoft.Extensions.Hosting.Internal.HostingLoggerExtensions.Starting||Hosting starting|
2022-01-18 09:14:13.05861-05:00|DEBUG|Microsoft.AspNetCore.Mvc.ModelBinding.ModelBinderFactory..ctor||Registered model binder providers, in the following order: Microsoft.AspNetCore.Mvc.ModelBinding.Binders.BinderTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ServicesModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.BodyModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.HeaderModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FloatingPointTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.EnumTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.DateTimeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.SimpleTypeModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.CancellationTokenModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ByteArrayModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FormFileModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.FormCollectionModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.KeyValuePairModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.DictionaryModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ArrayModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.CollectionModelBinderProvider, Microsoft.AspNetCore.Mvc.ModelBinding.Binders.ComplexObjectModelBinderProvider|
2022-01-18 09:14:13.29288-05:00|DEBUG|Microsoft.AspNetCore.SignalR.Internal.DefaultHubProtocolResolver..ctor||Registered SignalR Protocol: json, implemented by Microsoft.AspNetCore.SignalR.Protocol.JsonHubProtocol.|
2022-01-18 09:14:13.34641-05:00|TRACE|Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher`1.DiscoverHubMethods||'LiveUpdateSource' hub method 'SendMessageAsync' is bound.|
Segmentation fault
```

### To reproduce
None

### Expected behavior
Application should start in container.

### Further technical details
Microsoft.Data.SqlClient version:4.0.1
.NET target: 6 (upgrading from Core 3.1)
SQL Server version: 2022
Operating system: Docker container

**Additional context**
There is a temporary solution that works, by calling Initialize() as the first thing in Program.cs
```
public static class NetSecurityNativeFix
{
    public static void Initialize()
    {
        var result = -1;
        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
        {
            result = NetSecurityNative_EnsureGssInitialized();
        }
        if (result is 0)
        {
            Console.WriteLine($"{nameof(NetSecurityNativeFix)}::{nameof(Initialize)} EnsureGssInitialized workaround succeeded");
        }
        else
        {
            Console.WriteLine($"{nameof(NetSecurityNativeFix)}::{nameof(Initialize)} EnsureGssInitialized has failed result {result}");
        }
    }

    [System.Runtime.InteropServices.DllImport("System.Net.Security.Native")]
    [System.Runtime.InteropServices.DefaultDllImportSearchPaths(System.Runtime.InteropServices.DllImportSearchPath.UserDirectories)]
    private static extern int NetSecurityNative_EnsureGssInitialized();
}
```

**Comment by meghma** [comment]
I will retest

**Comment by meghma** [comment]
Re-tested and confirming the segmentation fault issue did not get resolved.

**Comment by meghma** [comment]
@JRahnama Hello - do you have any update about this issue. If you believe you have solved the problem, could you please suggest who else could we contact since the issue persists for us even after using  v 4.0.1?

**Comment by JRahnama** [comment]
@meghma I just retested the fix and was able to get value from server using Kerberso authentication. The fix is confirmed and switching back to 4.0.0 will cause the previous issue as expected. Not sure why you are facing this issue. There might be a possibility that some part of your code or container runs net security native from run-time and reuses that from the cache when it gets to SqlClient maybe? ( we have implemented this temporarily till next patch release for dotnet runtime gets released).
